src/IconBase.tsx

```diff
@@ -0,0 +1,60 @@
+import React, { SVGProps } from "react";
+
+export type IconSize = "xs" | "sm" | "md" | "lg" | "xl" | "xxl" | "xxxl";
+
+export interface IconProps
+  extends Omit<SVGProps<SVGSVGElement>, "onLoad" | "onError" | "ref"> {
+  /** Size (width and height) of the icon. Defaults to "md" or 16x16px */
+  size?: IconSize;
+  /** Render the title element with the provided text.
+   * More info: https://developer.mozilla.org/en-US/docs/Web/SVG/Element/title
+   */
+  title?: string;
+  /** Color of the icon. Defaults to "currentColor" */
+  color?: string;
+}
+
+function getSvgSize(size: IconSize) {
```

Comment:
There will be cases (I'm 100% sure) where we need custom sizes.

Reply:
This is the same API as we have for the current Icon component, and it doesn't look like we needed any custom sizes so far. Plus, as an edge case, those can be tweaked via `className`.